lib/api/services/wallet_service.dart

```diff
@@ -1,210 +1,149 @@
-              name: '',
-              phrase: keygen.phrase,
-              privateKey: keygen.privateKey,
-              publicKey: keygen.publicKey,
-              address: walletAddressResponse.address!,
-              blockchain: Blockchain(
-                  name: blockchainResponseModel.blockchainData!.name!,
-                  ticker: blockchainResponseModel.blockchainData!.ticker!,
-                  unit: blockchainResponseModel.blockchainData!.unit!,
-                  precision: blockchainResponseModel.blockchainData!.precision!,
-                  logo: blockchainResponseModel.blockchainData!.logo!,
-                  network_fee: blockchainResponseModel.blockchainData!.blockchainFee!
-              ),
-              balance: 0,
-            );
-
-            return wallet;
-
-
-          }else{
-            throw Exception('${blockchainResponse.body}');
-          }
-
-
-        }else{
-          throw Exception('${addressResponse.body}');
-        }
+        Wallet wallet = Wallet(
+          name: '',
+          phrase: mnemonic,
+          privateKey: const HexEncoder().convert(privateKey.toBytes()),
+          publicKey: const HexEncoder().convert(publicKey.toBytes()),
+          address: address,
+          blockchain: Blockchain(
+              name: blockchainResponseModel.blockchainData.name,
+              ticker: blockchainResponseModel.blockchainData.ticker,
+              unit: blockchainResponseModel.blockchainData.unit,
+              precision: blockchainResponseModel.blockchainData.precision,
+              logo: blockchainResponseModel.blockchainData.logo,
+              network_fee:
+                  blockchainResponseModel.blockchainData.blockchainFee),
+          balance: 0,
+        );
 
-      }else {
-        throw Exception('${keygenResponse.body}');
+        return wallet;
+      } else {
+        throw Exception(blockchainResponse.body);
       }
-
-    }on Exception catch (e){
+    } on Exception catch (e) {
       throw Exception('ERROR : ${e.toString()}');
     }
-
   }
 
-  // @POST("/v1/recover") and @POST("/v1/wallet") and @POST("v1/blockchain") and @POST("/v1/balance")
-  Future<Wallet> recoverWallet(String phrase)async{
-    try{
+  Future<Wallet> recoverWallet(String phrase) async {
+    var seed = mnemonicToSeed(phrase);
+    var privateKey = PrivateKey.fromSeed(seed);
+    var publicKey = privateKey.getG1();
+    var puzzle = walletPuzzle.curry([Program.atom(publicKey.toBytes())]);
+    var puzzleHash = puzzle.hash();
+    var address = segwit.encode(Segwit('xch', puzzleHash));
 
-      final recoverKeyResponse = await http.post(
-        Uri.parse('${baseURL}/v1/recover'),
+    try {
+      final blockchainResponse = await http.post(
+        Uri.parse('$baseURL/v2/blockchain'),
         headers: <String, String>{
           'Content-Type': 'application/json; charset=UTF-8',
         },
         body: jsonEncode(<String, String>{
-          'phrase': phrase,
+          'blockchain': 'xch',
```

Comment:
Hmm since we are now referencing `xch` in multiple places, probably need to have it some sort of a global const string.